service-base/src/main/java/org/eclipse/hono/service/plan/PrometheusBasedResourceLimitChecks.java

```diff
@@ -126,47 +141,60 @@ public PrometheusBasedResourceLimitChecks(final WebClient webClient) {
+     * have been reached.
+     * @param tenant The tenant configuration to check the limit against.
+     * @param payloadSize The message payload size in bytes.
+     * @return A future indicating the outcome of the check.
+     *         <p>
+     *         The future will be failed with a {@link ServiceInvocationException}
+     *         if the check could not be performed.
      */
-    public String getHost() {
-        return host;
-    }
+    public Future<Boolean> isMessageLimitReached(final TenantObject tenant,
+            final long payloadSize) {
 
-    /**
-     * Sets the host of the Prometheus server to retrieve metrics from.
-     * <p>
-     * The default value of this property is {@code null}.
-     *
-     * @param host The host name or IP address.
-     */
-    public void setHost(final String host) {
-        this.host = Objects.requireNonNull(host);
-    }
+        Objects.requireNonNull(tenant);
+        final long maxBytes = getMaximumNumberOfBytes(tenant);
+        final LocalDate effectiveSince = getEffectiveSince(tenant);
+        final long periodInDays = getPeriodInDays(tenant);
 
-    /**
-     * Gets the port of the Prometheus server to retrieve metrics from.
-     *
-     * @return port The port number.
-     */
-    public int getPort() {
-        return port;
-    }
+        log.trace("message limit config for tenant [{}] are [{}:{}, {}:{}, {}:{}]", tenant.getTenantId(),
+                FIELD_MAX_BYTES, maxBytes, FIELD_EFFECTIVE_SINCE, effectiveSince, FIELD_PERIOD_IN_DAYS, periodInDays);
 
-    /**
-     * Sets the port of the Prometheus server to retrieve metrics from.
-     * <p>
-     * The default value of this property is 9090.
-     *
-     * @param port The port number.
-     * @throws IllegalArgumentException if the port number is &lt; 0 or &gt; 2^16 - 1
-     */
-    public void setPort(final int port) {
-        if (PortConfigurationHelper.isValidPort(port)) {
-            this.port = port;
+        if (maxBytes == -1 || effectiveSince == null || periodInDays <= 0 || payloadSize <= 0) {
+            return Future.succeededFuture(Boolean.FALSE);
         } else {
-            throw new IllegalArgumentException("invalid port number");
+            final long dataUsagePeriod = calculateDataUsagePeriod(effectiveSince, periodInDays);
+            if (dataUsagePeriod <= 0) {
+                return Future.succeededFuture(Boolean.FALSE);
+            }
+
+            final String queryParams = String.format("floor(sum(increase(%s{status=\"%s\", tenant=\"%s\"} [%sd])))",
+                    MESSAGES_PAYLOAD_SIZE_METRIC_NAME,
+                    MetricsTags.ProcessingOutcome.FORWARDED.asTag().getValue(),
+                    tenant.getTenantId(),
+                    dataUsagePeriod);
+
+            final ExpiringValueCache<Object, Object> cache = Optional
+                    .ofNullable(cacheProvider)
+                    .map(provider -> provider.getCache(tenant.getTenantId()))
```

Comment:
If I am not mistaken, this means that we create a separate cache for each tenant, which will then contain a *single* entry with key `BYTES_CONSUMED_CACHE_ID` each, right?
FMPOV we should need a single cache instance only (e.g. with name `bytes_consumed`) which we can use for *all* tenants by using the tenant identifier as the key, don't you think?

Reply:
I created this having some more resource limits in the future such as _connection-time_ in the mind. I think it makes sense to create once cache instance (for `bytes_consumed`) and have `tenant-id` as the key.

Reply:
I see, then maybe we name the cache instance `limits` and use e.g. `${tenant-id}_bytes_consumed` as the key? We can then later extend with e.g. `${tenant-id}_connection_time`. WDYT?

Reply:
sounds good.